src/components/DateRangeInput/DateRangeInput.tsx

```diff
@@ -0,0 +1,316 @@
+    return [now, now];
+  }
+  return value.map(v => dayjs(v)) as [Dayjs, Dayjs];
+};
+
+const DateRangeInput: React.FC<
+  DateRangeInputProps & Omit<TextInputProps, 'value' | 'onChange' | 'label' | 'placeholder'>
+> = ({
+  value,
+  format = 'MM/DD/YYYY',
+  withTime,
+  withPresets,
+  onChange = noop,
+  labelStart,
+  labelEnd,
+  placeholderStart,
+  placeholderEnd,
+  ...rest
+}) => {
+  const datesFormatted = parseDateValue(value);
+  const [currentDateRange, setCurrentRange] = useState(value);
+  const [currentMonth, setCurrentMonth] = useState(datesFormatted[0]);
+  const [prevDateRange, setPrevDateRange] = useState(value);
+  const [open, setOpen] = useState(false);
+
+  const nextMonth = currentMonth.add(1, 'month');
+
+  const onCancel = useCallback(() => {
+    setCurrentRange(prevDateRange);
+    setOpen(false);
+  }, [setOpen, prevDateRange, setCurrentRange]);
+
+  const onApply = useCallback(
+    e => {
+      e.preventDefault();
+      setPrevDateRange(currentDateRange);
+      onChange(currentDateRange);
+      setOpen(false);
+    },
+    [setOpen, setPrevDateRange, onChange, currentDateRange]
+  );
+
+  const onNextMonth = useCallback(
+    e => {
+      e.preventDefault();
+      const next = currentMonth.add(1, 'month');
+      setCurrentMonth(next);
+    },
+    [currentMonth, setCurrentMonth]
+  );
+  const onPreviousMonth = useCallback(
+    e => {
+      e.preventDefault();
+      const next = currentMonth.subtract(1, 'month');
+      setCurrentMonth(next);
+    },
+    [currentMonth, setCurrentMonth]
+  );
+
+  const formatDate = useCallback(
+    (values, key) => {
+      if (!values || !Array.isArray(values)) {
+        return '';
+      }
+      return values[key] ? dayjs(values[key]).format(format) : '';
+    },
+    [format]
+  );
+  const onExpand = useCallback(
+    e => {
+      e.preventDefault();
+      setOpen(true);
+    },
+    [setOpen]
+  );
+
+  const onDaySelect = useCallback(
+    (dateChanged: Dayjs) => {
+      if (!currentDateRange || currentDateRange?.length === 0) {
+        return setCurrentRange([dateChanged.toDate()]);
+      }
+
+      const start = dayjs(currentDateRange[0]);
+      if (dateChanged.isBefore(start, 'date')) {
+        return setCurrentRange([dateChanged.hour(start.hour()).minute(start.minute()).toDate()]);
+      }
+
+      if (currentDateRange[1]) {
+        const end = dayjs(currentDateRange[1]);
+        if (dateChanged.isAfter(start, 'date') && dateChanged.isBefore(end, 'date')) {
+          return setCurrentRange([
+            dateChanged.hour(start.hour()).minute(start.minute()).toDate(),
+            currentDateRange[1],
+          ]);
+        }
+      }
+
+      return setCurrentRange([currentDateRange[0], dateChanged.toDate()]);
+    },
+    [setCurrentRange, currentDateRange]
+  );
+
+  const isDisabled = useCallback(() => !currentDateRange || currentDateRange?.length < 2, [
+    currentDateRange,
+  ]);
+
+  const onPresetSelect = useCallback(
```

Comment:
Is there a chance we can "move" the month  screen depending on thhe selection? For example, if we are in the

"September - October" view and I  select "last 2 weeks" is it easy for the view to go to "August - September" with the days highlighted?

If it's not trivial  we can mark it as  an enhancement, but it helps people not get lost ( i.e. I was browsing months so I reached December 2020, I clicked "last 24h" and I didn't see anything change so I thought I found a bug. If the calendar month automatically went to today's month, I'd  be able to understand that the option has been correctly selected)

Reply:
It's done, selected months are always in view as presets are changing.